cognite/client/_api/datapoints.py

```diff
@@ -517,6 +498,248 @@ def __init__(self, config: ClientConfig, api_version: str | None, cognite_client
         self._RETRIEVE_LATEST_LIMIT = 100
         self._POST_DPS_OBJECTS_LIMIT = 10_000
 
+        self.query_validator = _DpsQueryValidator(dps_limit_raw=self._DPS_LIMIT_RAW, dps_limit_agg=self._DPS_LIMIT_AGG)
+
+    @overload
+    def __call__(
+        self,
+        queries: DatapointsQuery,
+        *,
+        return_arrays: Literal[True],
+    ) -> Iterator[DatapointsArray]: ...
+
+    @overload
+    def __call__(
+        self,
+        queries: Sequence[DatapointsQuery],
+        *,
+        return_arrays: Literal[True],
+    ) -> Iterator[DatapointsArrayList]: ...
+
+    @overload
+    def __call__(
+        self,
+        queries: DatapointsQuery,
+        *,
+        return_arrays: Literal[False],
+    ) -> Iterator[Datapoints]: ...
+
+    @overload
+    def __call__(
+        self,
+        queries: Sequence[DatapointsQuery],
+        *,
+        return_arrays: Literal[False],
+    ) -> Iterator[DatapointsList]: ...
+
+    def __call__(
+        self,
+        queries: DatapointsQuery | Sequence[DatapointsQuery],
+        *,
+        start: int | str | datetime.datetime | None = None,
+        end: int | str | datetime.datetime | None = None,
+        aggregates: Aggregate | str | list[Aggregate | str] | None = None,
+        granularity: str | None = None,
+        timezone: str | datetime.timezone | ZoneInfo | None = None,
+        target_unit: str | None = None,
+        target_unit_system: str | None = None,
+        ignore_unknown_ids: bool = False,
+        include_status: bool = False,
+        ignore_bad_datapoints: bool = True,
+        treat_uncertain_as_bad: bool = True,
+        chunk_size_datapoints: int = 100_000,
+        chunk_size_time_series: int | None = None,
+        return_arrays: bool = True,
+    ) -> Iterator[DatapointsArray | DatapointsArrayList | Datapoints | DatapointsList]:
```

Comment:
These are all essentially default values as I understand it. Not sure about this suggestion, but what about having a factory method on  `DatapointQuery` that would achieve the same?

Then, you can simplify the signature of this function, which makes is more readable:
```suggestion
    def __call__(
        self,
        queries: DatapointsQuery | Sequence[DatapointsQuery],
        *,
        chunk_size_datapoints: int = 100_000,
        chunk_size_time_series: int | None = None,
        return_arrays: bool = True,
    ) -> Iterator[DatapointsArray | DatapointsArrayList | Datapoints | DatapointsList]:
```
If you do this reduction in parameter, we can use it as an experiment in training the user to use the `DatapointQuery` class instead as we are forcing them to do so. In case it turns out to be too complicated for endusers and we get a feedback, we can always introduce the default arguments later. Much easier to add then remove :)

Reply:
The problem with this approach is that you have to give the same parameters to all queries, instead of passing once at top-level. I am uncertain, but I think I favor keeping this method similar to the existing ones to make it feel as familiar as possible

Reply:
My thinking is that you would have a factory method on the `DatapointsQuery` that allowed instansiate multiple queries with the same parameters. Something like shown below. The thinking is to make it clearer what is query parameters, and what are parameters about how to query. 

```python
DatapointsQuery.create_multiple(
        id: int | list[int],
        external_id: str | list[str],
        instance_id: NodeId | list[NodeId],
        start: int | str | datetime.datetime | None = None,
        end: int | str | datetime.datetime | None = None,
        aggregates: Aggregate | str | list[Aggregate | str] | None = None,
        granularity: str | None = None,
        timezone: str | datetime.timezone | ZoneInfo | None = None,
        target_unit: str | None = None,
        target_unit_system: str | None = None,
        ignore_unknown_ids: bool = False,
        include_status: bool = False,
        ignore_bad_datapoints: bool = True,
        treat_uncertain_as_bad: bool = True,
)
```

Reply:
Sure, I'll do it! I really like the short signature:
```py
        self,
        queries: DatapointsQuery | Sequence[DatapointsQuery],
        *,
        chunk_size_datapoints: int = 100_000,
        chunk_size_time_series: int | None = None,
        return_arrays: bool = True,
```